### PR TITLE
log edge func report with path and duration

### PR DESCRIPTION
**Pull request by schehata**
In order for the edge functions to appear on Axiom dashboard a report must be sent. This PR prints a simple report to console, this would be parsed by Axiom drain and extract the report information.

- [X] log edge report
- [ ] exclude middleware from edge report?

### CODE REVIEW

File: src/withAxiom.ts [hunk 1]
```diff
@@ -101,20 +102,47 @@ function withAxiomNextApiHandler(handler: NextApiHandler): NextApiHandler {
...
+      ip: req.ip,
+      region: req.geo?.region,
+      host: req.nextUrl.host,
+      method: req.method,
+      path: req.nextUrl.pathname,
+      scheme: req.nextUrl.protocol.replace(':', ''),
+      statusCode: 0,
+      userAgent: req.headers.get('user-agent'),
+    };
+    const axiomRequest = req as AxiomRequest;
```
Comment (bahlo) [inline src/withAxiom.ts:124]: This is missing the logger, isn't it?

File: src/withAxiom.ts [hunk 2]
```diff
@@ -101,20 +102,47 @@ function withAxiomNextApiHandler(handler: NextApiHandler): NextApiHandler {
...
   return async (req, ev) => {
+    const report = {
+      startTime: new Date().getTime(),
+      ip: req.ip,
+      region: req.geo?.region,
+      host: req.nextUrl.host,
+      method: req.method,
+      path: req.nextUrl.pathname,
+      scheme: req.nextUrl.protocol.replace(':', ''),
+      statusCode: 0,
```
Comment (bahlo) [inline src/withAxiom.ts:121]:
```suggestion
```

File: src/withAxiom.ts [hunk 3]
```diff
@@ -101,20 +102,47 @@ function withAxiomNextApiHandler(handler: NextApiHandler): NextApiHandler {
...
+      userAgent: req.headers.get('user-agent'),
+    };
+    const axiomRequest = req as AxiomRequest;
+
     try {
-      const res = await handler(req, ev);
+      const res = await handler(axiomRequest, ev);
+      report.statusCode = res?.status || 0;
```
Comment (bahlo) [inline src/withAxiom.ts:128]:
```suggestion
      if (res) {
        report.statusCode = res.status
      }
```

**Review by bahlo — APPROVED** [review]

File: src/withAxiom.ts (hunk 1 above)
Comment (schehata) [reply src/withAxiom.ts:123]: well, this MR only handles printing the edge report. attach request information to log events is in a different [PR](https://github.com/axiomhq/next-axiom/pull/28), I am still working on it though.

**Pull request merged**